ProcessMaker/Jobs/DownloadSecurityLog.php

```diff
@@ -37,18 +50,161 @@ public function __construct(User $user, string $format, int $userId = null)
+            event(new SecurityLogDownloadJobCompleted($this->user, false, $message));
+        }
+    }
+
+    /**
+     * Generate the content with the corresponding format
+     */
+    protected function generateContent(array $collection)
+    {
+        return $this->format === static::FORMAT_CSV ? $this->toCSV($collection) : $this->toXML($collection);
+    }
+
+    /**
+     * To CSV
+     */
+    protected function toCSV(array $collection)
+    {
+        $first = current($collection);
+        $headers = array_keys($first ?: []);
+        $content = '';
+        $line = [];
+        foreach ($headers as $key) {
+            $line[] = $key;
+        }
+        $content .= implode(static::CSV_SEPARATOR, $line);
+        foreach ($collection as $item) {
+            $line = [];
+            foreach ($headers as $key) {
+                if (is_object($item[$key])) {
+                    $line[] = json_encode($item[$key]);
+                } else {
+                    $line[] = isset($item[$key]) ? $item[$key] : null;
+                }
+            }
+            $content .= PHP_EOL . implode(static::CSV_SEPARATOR, $line);
         }
+        return $content;
+    }
+
+    /**
+     * To XML
+     */
+    protected function toXML(array $collection)
+    {
+        $content = '<?xml version="1.0" encoding="UTF-8"?>' . PHP_EOL;
+        $content .= '<securityLogs>';
+        $tab = "\t";
+        foreach ($collection as $item) {
+            $content .= PHP_EOL . $tab . '<securityLog>';
+            foreach ($item as $key => $value) {
+                if (is_object($value)) {
+                    $value = json_encode($value);
+                }
+                $content .= sprintf(
+                    '%s<%s>%s</%s>',
+                    PHP_EOL . $tab,
+                    $key,
+                    $value,
+                    $key
+                );
+            }
+            $content .= PHP_EOL . $tab . '</securityLog>';
+        }
+        $content .= PHP_EOL . '</securityLogs>';
+
+        return $content;
+    }
+
+    /**
+     * Get collection
+     *
+     * @return array
+     */
+    protected function getCollection()
+    {
+        $query = SecurityLog::query();
+        if ($this->userId) {
+            $query->where('user_id', $this->userId);
+        }
+        $results = [];
+        $query->chunk(1000, function (Collection $collection) use (&$results) {
+            foreach ($collection as $item) {
+                $results[] = $item->toArray();
+            }
+            unset($item);
+        });
+
+        return $results;
+    }
+
+    /**
+     * Create temp url
+     *
+     * @param string $filename
+     * @param string $content
+     * @param Carbon $expires
+     *
+     * @return string
+     */
+    protected function createTemporaryUrl(string $filename, string $content, Carbon $expires)
+    {
+        $disk = Storage::disk('s3');
+        $disk->put($filename, $content, [
```

Comment:
If the size is large consider to use strem logic like this:
Reference: https://freek.dev/235-upload-large-files-to-s3-using-laravel-5
```
use Illuminate\Support\Facades\DB;
use Illuminate\Support\Facades\Storage;

public function export() {
    // Get a disk manager for S3
    $disk = Storage::disk('s3');

    // Create a stream
    $stream = fopen('php://temp', 'w+');

    // Use a cursor to iterate over the table data
    DB::table('security_logs')->orderBy('id')->cursor()->each(function ($record) use ($stream) {
        // Convert each record to an array and write it to the stream
        fputcsv($stream, (array) $record);
    });

    // Rewind the stream
    rewind($stream);

    // Save the stream to S3
    $filename = 'exports/my_table.csv';
    $disk->put($filename, stream_get_contents($stream));

    // Close the stream
    fclose($stream);
}
```